Userland/Libraries/LibAudio/FlacLoader.cpp

```diff
@@ -0,0 +1,899 @@
+        CHECK_ERROR_STRING;
+        current_subframes.append(move(subframe_samples));
+    }
+
+    bit_stream.align_to_byte_boundary();
+
+    // TODO: check checksum
+    [[maybe_unused]] u16 footer_checksum = bit_stream.read_bits_big_endian(16);
+
+    Vector<i32> left, right;
+
+    switch (channel_type) {
+    case FlacFrameChannelType::Mono:
+        left = right = current_subframes[0];
+        break;
+    case FlacFrameChannelType::Stereo:
+    // TODO mix together surround channels on each side?
+    case FlacFrameChannelType::StereoCenter:
+    case FlacFrameChannelType::Surround4p0:
+    case FlacFrameChannelType::Surround5p0:
+    case FlacFrameChannelType::Surround5p1:
+    case FlacFrameChannelType::Surround6p1:
+    case FlacFrameChannelType::Surround7p1:
+        left = current_subframes[0];
+        right = current_subframes[1];
+        break;
+    case FlacFrameChannelType::LeftSideStereo:
+        // channels are left (0) and side (1)
+        left = current_subframes[0];
+        right.ensure_capacity(left.size());
+        for (size_t i = 0; i < left.size(); ++i) {
+            // right = left - side
+            right.unchecked_append(left[i] - current_subframes[1][i]);
+        }
+        break;
+    case FlacFrameChannelType::RightSideStereo:
+        // channels are side (0) and right (1)
+        right = current_subframes[1];
+        left.ensure_capacity(right.size());
+        for (size_t i = 0; i < right.size(); ++i) {
+            // left = right + side
+            left.unchecked_append(right[i] + current_subframes[0][i]);
+        }
+        break;
+    case FlacFrameChannelType::MidSideStereo:
+        // channels are mid (0) and side (1)
+        left.ensure_capacity(current_subframes[0].size());
+        right.ensure_capacity(current_subframes[0].size());
+        for (size_t i = 0; i < current_subframes[0].size(); ++i) {
+            i64 mid = current_subframes[0][i];
+            i64 side = current_subframes[1][i];
+            mid *= 2;
+            // prevent integer division errors
+            left.unchecked_append(static_cast<i32>((mid + side) / 2));
+            right.unchecked_append(static_cast<i32>((mid - side) / 2));
+        }
+        break;
+    }
+
+    VERIFY(left.size() == right.size());
+
+    // TODO: find the correct rescale offset
+    double sample_rescale = static_cast<double>(1 << pcm_bits_per_sample(m_current_frame->bit_depth));
+    dbgln_if(AFLACLOADER_DEBUG, "Sample rescaled from {} bits: factor {:.1f}", pcm_bits_per_sample(m_current_frame->bit_depth), sample_rescale);
+
+    m_current_frame_data.clear_with_capacity();
+    m_current_frame_data.ensure_capacity(left.size());
+    // zip together channels
+    for (size_t i = 0; i < left.size(); ++i) {
+        Frame frame = { left[i] / sample_rescale, right[i] / sample_rescale};
+        // if (left[i] != 0)
+            // dbgln("Frame data {:8} : {:8} -> {:7.4f} {:7.4f}", left[i], right[i], frame.left, frame.right);
```

Comment:
This doesn't seem very useful
if it's to aid debugging, just put it behind `if constexpr (AFLACLOADER_DEBUG)`

Reply:
I forgot the commented-out code in several places